src/main/kotlin/com/exactpro/th2/codec/AbstractCodecProcessor.kt

```diff
@@ -124,6 +128,23 @@ abstract class AbstractCodecProcessor<BATCH, GROUP, MESSAGE>(
             return null
         }
 
+        var mismatchDetested = false
+        messageGroup.groupItems.forEach { message ->
+            val messageBook: String = message.book(batch)
+            val eventBook: String? = message.eventBook
+            if (eventBook?.equals(messageBook) == false) {
+                eventProcessor.onErrorEvent(
+                    "Book name mismatch in '$messageBook' message and '$eventBook' parent event ids",
+                    message.eventId,
+                    listOf(message.id(batch))
+                )
```

Comment:
I am missing something probably, but if we just checked that the book in the parent event attached to the message is different from the book in the message itself why do we try to publish an event with that parent event ID and the message ID attached? Won't it cause the same problem we have right now?

Reply:
I see now, thank you